addons/udes_sale_stock/models/sale_order.py

```diff
@@ -208,6 +208,40 @@ def confirm_if_due(self):
         )
         return so_to_confirm.action_confirm()
 
+    def get_current_demand(self, products=None):
+        """ Get current demand created by Sale Orders per product - regardless
+        of expected delivery date """
+        OrderLine = self.env['sale.order.line']
+
+        # Get order lines
+        offset = 0
+        limit = 100
+        domain = [('state', '=', 'sale'),
+                  ('is_cancelled', '=', False)]
+        if products:
+            domain.append(('product_id', 'in', products.ids))
+        batch = OrderLine.search(domain, offset=offset, limit=limit)
+        demand = defaultdict(int)
+
+        while batch:
+            # Cache stuff
+            batch.mapped('move_ids')
+            for line in batch:
+                # If any of the mls are done or assigned then skip this line
```

Comment:
Typo: `mls`  --> `moves` (this might come from the original code)

`done or assigned`  vs skipped states done/cancel